Aws client updates

martinheidegger opened the pull request:
Minor updates to the AWS client code. See the commit messages for detail reasoning.

mvayngrib commented on package.json:
```diff
@@ -20,7 +20,7 @@
     "@types/sinon": "^4.1.2",
     "@types/superagent": "^3.5.7",
-    "aws-sdk": "^2.631.0",
+    "aws-sdk": "^2.1056.0",
```
should we use 2.1001.0 to match the lambda runtime environment? https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

martinheidegger replied:
Updated and adding script to aws-utils for testing: https://github.com/tradle/aws-utils/compare/bd1686b709d555366bd2729420fe8f1b3bfc06e4...e155df7564ec64815a1d82f94d68b5a41d69f9d4

urbien commented:
same question regarding SES

martinheidegger commented:
I ran the script yesterday to update the generated data. The script in `aws-utils` uses geographical distance to [identify `us-west-1`](https://github.com/tradle/aws-utils/blob/acc2672fcd1e75d73f855586c0d300ea894f0f85/packages/aws-common-utils/src/countries.ts#L1911) using [geographic distance](https://github.com/tradle/aws-utils/blob/acc2672fcd1e75d73f855586c0d300ea894f0f85/scripts/generate_geodata/cli.js#L63) as sao-paolo is much further away. SES and SNS use the same data source/dependencies. To use Sao Paolo we would need to change the mapping systems by which to choose the data center to use for calling an address.

urbien commented:
what mapping system are you referring to?
see that San Paolo is in the list of regions that has SNS capability
https://docs.aws.amazon.com/sns/latest/dg/sns-supported-regions-countries.html

martinheidegger commented:
I am referring to the system for mapping  country ←→ aws-datacenter. Currently we are using a geographic mapping system that chooses datacenter by geographic proximity. (see code referenced above).

urbien commented:
right, but that is only a fallback for the case when current region does not have SNS capability, correct? And Brazil acquired that  capability in 2020 (after the geo list was created).

martinheidegger commented:
For brazil, the sao-paolo datacenter is used as it is closest ([the aws documentation](https://github.com/tradle/aws-utils/blob/acc2672fcd1e75d73f855586c0d300ea894f0f85/scripts/generate_geodata/sms_data.js#L5) is used to generate the data), but since there is no datacenter in Mexico we need to fallback by a mechanism.

martinheidegger commented:
I updated the PR. It is now rebased to the current master and contains the recent updates in aws-util which contains the algorithm changes, of particular note: https://github.com/tradle/aws-utils/commit/1deb0d71036e5159089cba975528b828c9fd1dbc
